### PR TITLE
Adding JWT authentication filter to admin APIs

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs [hunk 1]
```diff
@@ -33,12 +33,20 @@ public AuthorizationLevelAttribute(AuthorizationLevel level)
                 throw new ArgumentNullException("actionContext");
             }
 
-            // determine the authorization level for the function and set it
-            // as a request property
-            var secretManager = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
+            AuthorizationLevel requestAuthorizationLevel = actionContext.Request.GetAuthorizationLevel();
+            
+            // If the request has not yet been authenticated, authenticate it
+            if (requestAuthorizationLevel == AuthorizationLevel.Anonymous)
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs]: This line is the relevant change in the `AuthorizationLevelAttribute`, with this change, we'll only go through the authentication process (using `code` query string or `x-functions-key` header if the request is not yet authenticated.

Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs]: I do want to break this up into a separate filter, so only the authorization logic is here, but will do that in a separate PR.

File: src/WebJobs.Script/ScriptConstants.cs [hunk 2]
```diff
@@ -48,6 +48,9 @@ public static class ScriptConstants
 
         public const string FeatureFlagDisableShadowCopy = "DisableShadowCopy";
 
+        public const string AdminJwtValidAudienceFormat = "https://{0}.scm.azurewebsites.net ";
```
Comment (mathewc) [inline src/WebJobs.Script/ScriptConstants.cs]: reverse these
